env/file_system.cc

```diff
@@ -83,6 +84,31 @@ FileOptions FileSystem::OptimizeForCompactionTableRead(
   return optimized_file_options;
 }
 
+std::string FileSystem::GenerateUniqueId(const IOOptions& opts,
+                                         IODebugContext* dbg) {
+  std::string uuid_file = "/proc/sys/kernel/random/uuid";
```

Comment:
How about Windows and MacOS?

Reply:
Windows has its own implementation, and the implementation in this PR should work for MacOS I think? Its taken from ```Env::GenerateUniqueId```.

Regarding your comment about this being OS functionality, its a fair point. I noticed that ```env_librados.cc``` overrides it with its own implementation, but that also relies on system clock and there's nothing file system specific. So this could be considered as part of Env functionality.

Reply:
> Windows has its own implementation, and the implementation in this PR should work for MacOS I think? Its taken from `Env::GenerateUniqueId`.
> 
Verified on MacOS, and it does not work since `/proc/sys/kernel/random/uuid` does not exist (MacOS does not have /proc). This issue can be addressed in a separate PR. This PR itself LGTM.

Reply:
Never mind, the function handles the case when the /proc is missing, thus should be fine.